CoffeeChat/Controller/EditingViewController.swift

```diff
@@ -467,75 +504,82 @@ extension EditHeaderView: UISearchBarDelegate {
 private class EditFooterView: UIButton {
 
-    private var showingGroups = false
-    private var showingLess = false
+    private let showingGroups: Bool
+    private var showingLessRows = false
 
+    private let stack = UIStackView()
     private let label = UILabel()
     private let arrowView = UIImageView()
 
     var delegate: ((Bool) -> Void)?
 
-    convenience init(showsGroups: Bool) {
-        self.init(frame: .zero)
-        label.font = UIFont._16CircularStdMedium
-        label.textColor = .greenGray
+    init(showsGroups: Bool) {
         showingGroups = showsGroups
-        configure(showsGroups: showingGroups)
+        super.init(frame: .zero)
+
+        setupViews()
+        setupConstraints()
+        configure()
     }
 
-    override init(frame: CGRect) {
-        super.init(frame: frame)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
 
+    private func setupViews() {
+        label.font = UIFont._16CircularStdMedium
+        label.textColor = .greenGray
         label.numberOfLines = 0
         label.textAlignment = .center
         label.isUserInteractionEnabled = false
-        addSubview(label)
 
-        arrowView.image = UIImage(named: "right_arrow")
+        arrowView.image = UIImage(named: "rightArrow")
         arrowView.isUserInteractionEnabled = false
         arrowView.transform = CGAffineTransform(rotationAngle: CGFloat(Double.pi / 2))
-        arrowView.snp.makeConstraints { make in
-            make.size.equalTo(CGSize(width: 9, height: 18))
-        }
 
-        let stack = UIStackView()
         stack.isUserInteractionEnabled = false
         stack.alignment = .center
         stack.axis = .horizontal
         stack.spacing = 12
-        stack.distribution = .fillProportionally
         stack.insertArrangedSubview(arrowView, at: 0)
         stack.insertArrangedSubview(label, at: 1)
         addSubview(stack)
+
+        addTarget(self, action: #selector(buttonPressed), for: .touchDown)
+    }
+
+    private func setupConstraints() {
+        arrowView.snp.makeConstraints { make in
+            make.size.equalTo(CGSize(width: 9, height: 18))
+        }
+
         stack.snp.makeConstraints { make in
             make.center.equalToSuperview()
             make.width.equalToSuperview().inset(44)
         }
-
-        addTarget(self, action: #selector(buttonAction), for: .touchDown)
     }
 
-    func changeViewState(less: Bool) {
-        showingLess = less
-        label.text = less
-            ? "View your other \(showingGroups ? "groups" : "interests")"
-            : "View fewer \(showingGroups ? "groups" : "interests")"
-        arrowView.transform = less
-            ? CGAffineTransform(rotationAngle: CGFloat(-Double.pi / 2))
-            : CGAffineTransform(rotationAngle: CGFloat(Double.pi / 2))
-        delegate?(showingLess)
-    }
+    func changeViewState(isShowingLess: Bool) {
+        showingLessRows = isShowingLess
```

Comment:
is it possible to just utilize the`showingLessRows` class var for all of the logic inside of this function, instead of using the parameter? it's a little confusing to follow, with `!showingLessRows` being passed when the function is called from `buttonPressed`, but not actually assigning the toggled value to the `showingLessRows` var until inside this function.